InnerEye/ML/run_ml.py

```diff
@@ -760,6 +761,8 @@ def create_ensemble_model_and_run_inference(self) -> None:
                 self.register_model(checkpoint_handler.get_best_checkpoints(), ModelProcessing.ENSEMBLE_CREATION)
 
         if not self.azure_config.only_register_model:
+            if is_global_rank_zero():
+                torch.cuda.empty_cache()
```

Comment:
I feel this should not be necessary. This only defragments the memory but at this point it should be "empty" whatever is there can be GC because there are no references.